src/pharmpy/tools/amd/run.py

```diff
@@ -913,63 +913,38 @@ def _run_mechanistic_exploratory_covariates(model, modelfit_results):
-                    structural_searchspace.append(cov_statement)
-                    mechanistic_searchspace.append(cov_statement)
-                    filtered_searchspace.append(cov_statement)
-                else:
-                    current_cov = []
-                    for cov in cov_statement.covariate:
-                        if cov in mechanistic_covariates:
-                            current_cov.append(cov)
-                    if current_cov:
-                        mechanistic_cov = Covariate(
-                            cov_statement.parameter,
-                            tuple(current_cov),
-                            cov_statement.fp,
-                            cov_statement.op,
-                            cov_statement.optional,
-                        )
-                        mechanistic_searchspace.append(mechanistic_cov)
-                        if filtered_cov_set := tuple(
-                            set(cov_statement.covariate) - set(current_cov)
-                        ):
-                            filtered_cov = Covariate(
-                                cov_statement.parameter,
-                                filtered_cov_set,
-                                cov_statement.fp,
-                                cov_statement.op,
-                                cov_statement.optional,
-                            )
-                            filtered_searchspace.append(filtered_cov)
-                    else:
-                        filtered_searchspace.append(cov_statement)
+            mechanistic_searchspace, filtered_searchspace = _mechanistic_cov_extraction(
+                search_space, model, mechanistic_covariates
+            )
 
+            # FIXME : Move to validation
             if not mechanistic_searchspace:
                 warnings.warn(
                     'No covariate effect for given mechanistic covariates found.'
                     ' Skipping mechanistic COVsearch.'
                 )
             else:
+                mechanistic_searchspace = ModelFeatures.create_from_mfl_statement_list(
+                    mechanistic_searchspace
+                )
+
                 res = run_tool(
                     'covsearch',
-                    mfl_stringify(mechanistic_searchspace),
+                    mechanistic_searchspace,
                     model=model,
                     strictness=strictness,
                     results=modelfit_results,
                     path=path / 'covsearch_mechanistic',
                 )
-                index_offset = int(
-                    res.summary_tool.iloc[-1].name[-1][13:]
-                )  # Get largest number of run
+                model_db = res.tool_database.model_database
+                all_models = [model_db.retrieve_model(model) for model in model_db.list_models()]
+                covsearch_model_number = [
+                    int(model.name[13:])
```

Comment:
I'm guessing the 13 here is the length of `covsearch_run`, I see this both as a bit confusing (it's not apparent what 13 comes from) and as a bit risky if we e.g. change the names of the candidates. Perhaps regex can be used instead to capture the number? This would maybe address the first concern and at least make the second concern explicit.

Reply:
Will use regex to find the number in the end of the name instead. Thanks for your input!